common/src/main/java/com/hortonworks/registries/common/util/HdfsFileStorage.java

```diff
@@ -33,41 +41,121 @@
  */
 public class HdfsFileStorage implements FileStorage {
 
+    private static final Logger LOG = LoggerFactory.getLogger(HdfsFileStorage.class);
+
     // the configuration keys
     public static final String CONFIG_FSURL = "fsUrl";
     public static final String CONFIG_DIRECTORY = "directory";
+    public static final String CONFIG_KERBEROS_PRINCIPAL = "hdfs.kerberos.principal";
+    public static final String CONFIG_KERBEROS_KEYTAB = "hdfs.kerberos.keytab";
 
-    private String fsUrl;
     private String directory = DEFAULT_DIR;
     private FileSystem hdfsFileSystem;
+    private UserGroupInformation userGroupInformation;
 
     @Override
-    public void init(Map<String, String> config) {
+    public void init(Map<String, String> config) throws IOException {
+        String fsUrl = null;
+        String kerberosPrincipal = null;
+        String keytabLocation = null;
         Configuration hdfsConfig = new Configuration();
         for(Map.Entry<String, String> entry: config.entrySet()) {
-            if(entry.getKey().equals(CONFIG_FSURL)) {
-                this.fsUrl = config.get(CONFIG_FSURL);
-            } else if(entry.getKey().equals(CONFIG_DIRECTORY)) {
-                this.directory = config.get(CONFIG_DIRECTORY);
-            } else {
-                hdfsConfig.set(entry.getKey(), entry.getValue());
+            switch (entry.getKey()) {
+                case CONFIG_FSURL:
+                    fsUrl = entry.getValue();
+                    break;
+                case CONFIG_DIRECTORY:
+                    directory = entry.getValue();
+                    break;
+                case CONFIG_KERBEROS_PRINCIPAL:
+                    kerberosPrincipal = entry.getValue();
+                    break;
+                case CONFIG_KERBEROS_KEYTAB:
+                    keytabLocation = entry.getValue();
+                    break;
+                default:
+                    hdfsConfig.set(entry.getKey(), entry.getValue());
             }
         }
 
         // make sure fsUrl is set
-        if(fsUrl == null) {
-            throw new RuntimeException("fsUrl must be specified for HdfsFileStorage.");
+        Preconditions.checkArgument(fsUrl != null, "fsUrl must be specified for HdfsFileStorage.");
+
+        Preconditions.checkArgument((keytabLocation == null) == (kerberosPrincipal == null),
```

Comment:
Either both should be `null` (no kerberos) or neither should be `null` (kerberos).

Reply:
Okay, condition is fine, just the description needs to be fixed then.
`Both ... to use kerberos or neither otherwise.` - or something smarter. It's just confusing a bit